### PR TITLE
[improve] Change Mysql Driver

### LINKED ISSUE #10225: [Improvement][Mysql] Change Mysql Driver
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

```
Loading class `com.mysql.jdbc.Driver'. 
This is deprecated. The new driver class is `com.mysql.cj.jdbc.Driver'.
The driver is automatically registered via the SPI and manual loading 
of the driver class is generally unnecessary.
```

We need to change to `com.mysql.cj.jdbc.Driver`.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by qingwli**
## Purpose of the pull request

Change Mysql Driver

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

close #10225

### CODE REVIEW

**Comment by songjianet** [comment]
Please associate the corresponding issue. @liqingwang

File: dolphinscheduler-master/src/main/resources/application.yaml [hunk 1]
```diff
@@ -135,8 +135,7 @@ spring:
     activate:
       on-profile: mysql
   datasource:
-    driver-class-name: com.mysql.jdbc.Driver
-    url: jdbc:mysql://127.0.0.1:3306/dolphinscheduler?useUnicode=true&characterEncoding=UTF-8
```
Comment (zhongjiajie) [inline dolphinscheduler-master/src/main/resources/application.yaml:139]: why should we delete attribute `url`?

Comment (qingwli) [reply dolphinscheduler-master/src/main/resources/application.yaml:139]: Activate profile only define the data source driver, and other fields, such as URL, username, and password we define at another place. You can see [this](https://github.com/apache/dolphinscheduler/issues/10103#issuecomment-1132669903) and [this](https://github.com/apache/dolphinscheduler/pull/10128/files)

Comment (zhongjiajie) [reply dolphinscheduler-master/src/main/resources/application.yaml:139]: Got it, Thanks for the clarification

**Review by zhongjiajie — APPROVED** [review]
LGTM, thanks

**Review by kezhenxu94 — APPROVED** [review]

**Pull request merged**